chore: Upgrade spectral dependency

filfreire opened the pull request:
## Changes

- Upgraded `@spotlight/spectral` dependency to latest versions. There's been some major changes on Spectral, and the latest version[ is no longer published in the same package](https://www.npmjs.com/package/@stoplight/spectral). Now we need to use specific packages directly, like `spectral-core`, `spectral-formats` and `spectral-rulesets`.
- Added a smoke check where we import our default Petstore OpenAPI example and we try to force an OpenAPI lint error.

## How to test locally

- Checkout this branch
- Run `npm run bootstrap`
- Run `npm run watch:app`
- On a 2nd terminal window run `npm run test:smoke:dev -- openapi`

filfreire committed:
chore: Upgrade spectral dependency

filfreire committed:
Add smoke check for Spectral changes

jackkav commented on packages/insomnia-inso/src/commands/lint-specification.ts:
```diff
@@ -50,9 +51,7 @@ export async function lintSpecification(
-  spectral.registerFormat('oas2', isOpenApiv2);
-  spectral.registerFormat('oas3', isOpenApiv3);
-  await spectral.loadRuleset('spectral:oas');
+  await spectral.setRuleset(oas as unknown as Ruleset);
```
could be a ts-ignore?

filfreire replied:
Yeah, seems to work 👍

filfreire committed:
Remove cast as unknown

filfreire committed:
Merge branch 'develop' into chore/upgrade-spectral-dep

jackkav reviewed (approved):
🚀

DMarby commented:
Does this mean that we can get rid of our `immer` dependency workaround as well? (ping @dimitropoulos) https://github.com/Kong/insomnia/blob/cc9f11be30f365fdc70cd96bab5fa48a591f2415/package.json#L68-L70

dimitropoulos commented on packages/insomnia-inso/src/commands/lint-specification.ts:
```diff
@@ -50,9 +51,8 @@ export async function lintSpecification(
-  spectral.registerFormat('oas2', isOpenApiv2);
-  spectral.registerFormat('oas3', isOpenApiv3);
-  await spectral.loadRuleset('spectral:oas');
+  // @ts-expect-error -- TSCONVERSION oas not being properly caught as Ruleset
```
please do not mark things as TSCONVERSION, that was a sortof one-time thing for the time when we converted to typescript.  all errors from here forward should be handled.  I'll try to take a look at this one.  I remember hitting it before when I upgraded spectral last year.

filfreire replied:
Understood. I went ahead and removed the `TSCONVERSION` keyword for this `@ts-expect-error` message.

dimitropoulos replied:
you removed the other one in spectral.ts it looks like (which also needed to be removed).

dimitropoulos replied:
fixed in 22fa1a3114c01dab4f3b2401d617abbc9ed68d4c

filfreire committed:
Rm immer dependency and edit ts-expect-error description

filfreire commented:
> Does this mean that we can get rid of our `immer` dependency workaround as well? (ping @dimitropoulos)
> 
> https://github.com/Kong/insomnia/blob/cc9f11be30f365fdc70cd96bab5fa48a591f2415/package.json#L68-L70

@DMarby @dimitropoulos I went ahead and also removed this `immer` dependency 👍

dimitropoulos committed:
fixes type errors

dimitropoulos committed:
updates rulesets to latest (released just 10 minutes ago)

https://github.com/stoplightio/spectral/commit/60770d1ecd7891b0d866417016d6b50b866e50eb

filfreire committed:
Merge remote-tracking branch 'origin' into chore/upgrade-spectral-dep

dimitropoulos committed:
experiment: does ts-node fix it

https://github.com/microsoft/TypeScript/issues/49257#issuecomment-1140632708

filfreire committed:
fix package-log

filfreire committed:
Merge branch 'develop' into chore/upgrade-spectral-dep
